### PR TITLE
Validate VK_EXT_pipeline_creation_cache_control

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Add VUIDs for VK_EXT_pipeline_creation_cache_control

VUID-VkGraphicsPipelineCreateInfo-pipelineCreationCacheControl-02878
VUID-VkComputePipelineCreateInfo-pipelineCreationCacheControl-02875
VUID-VkPipelineCacheCreateInfo-pipelineCreationCacheControl-02892
VUID-VkRayTracingPipelineCreateInfoNV-pipelineCreationCacheControl-02905
VUID-VkRayTracingPipelineCreateInfoKHR-pipelineCreationCacheControl-02905

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -3872,6 +3874,39 @@ bool CoreChecks::ValidatePipelineVertexDivisors(std::vector<std::shared_ptr<PIPE
     return skip;
 }
 
+bool CoreChecks::ValidatePipelineCacheControlFlags(VkPipelineCreateFlags flags, uint32_t index, const char *caller_name,
+                                                   const char *vuid) const {
+    bool skip = false;
+    if ((device_extensions.vk_ext_pipeline_creation_cache_control) &&
+        (enabled_features.pipeline_creation_cache_control_features.pipelineCreationCacheControl == VK_FALSE)) {
```
Comment (jeffbolznv) [inline layers/core_validation.cpp:3881]: Why do these need to check for the extension enabled? Shouldn't the error also fire if the extension is disabled?

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3872,6 +3874,39 @@ bool CoreChecks::ValidatePipelineVertexDivisors(std::vector<std::shared_ptr<PIPE
     return skip;
 }
 
+bool CoreChecks::ValidatePipelineCacheControlFlags(VkPipelineCreateFlags flags, uint32_t index, const char *caller_name,
```
Comment (jeffbolznv) [inline layers/core_validation.cpp:3877]: Can you call this for ray tracing as well?

Comment (sfricke-samsung) [reply layers/core_validation.cpp:3877]: hmmm, didn't see `VUID-VkRayTracingPipelineCreateInfoNV-pipelineCreationCacheControl-02905` and `VUID-VkRayTracingPipelineCreateInfoKHR-pipelineCreationCacheControl-02905`... will add

File: layers/core_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/core_validation.cpp:3881]: yes, good call, will remove below as well too

**Comment by mark-lunarg** [comment]
Running through CI.

**Pull request merged**